pkg/canary/scaler_reconciler_fixture_test.go

```diff
@@ -134,3 +147,34 @@ func newScalerReconcilerTestHPAV2() *hpav2.HorizontalPodAutoscaler {
 
 	return h
 }
+
+func newScaledObject() *keda.ScaledObject {
+	so := &keda.ScaledObject{
+		TypeMeta: metav1.TypeMeta{APIVersion: keda.SchemeGroupVersion.String()},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "podinfo",
+		},
+		Spec: keda.ScaledObjectSpec{
+			ScaleTargetRef: &keda.ScaleTarget{
+				Name: "podinfo",
+			},
+			PollingInterval: int32p(10),
+			MinReplicaCount: int32p(1),
+			MaxReplicaCount: int32p(4),
+			Triggers: []keda.ScaleTriggers{
+				{
+					Type: "prometheus",
+					Metadata: map[string]string{
+						"serverAddress": "http://flagger-prometheus.projectcontour:9090",
+						"metricName":    "http_requests_total",
+						"query":         `sum(rate(http_requests_total{deployment="podinfo-canary"}[2m]))`,
```

Comment:
Please replace `deployment` with app or something else, Flagger does not alter the original deployment and using `podinfo-canary` as the deployment name in tests is confusing to me.